### PR TITLE
Add superfluid support to Defillama

### PR DESCRIPTION
**Pull request by jo-chemla**
Add superfluid TVL from polygon and xdai

##### Twitter Link:
https://twitter.com/Superfluid_HQ

##### List of audit links if any:
[PeckShield-Audit](https://github.com/superfluid-finance/protocol-monorepo/blob/dev/packages/ethereum-contracts/audits/PeckShield-Audit-Report-Superfluid-v1.0.pdf)

##### Website Link:
https://www.superfluid.finance/

##### Logo (High resolution, preferably in .svg and .png, for application on both white and black backgrounds. Will be shown with rounded borders):
dark bg round: https://pbs.twimg.com/profile_images/1283443930890149889/K0bTNmr9_400x400.png
dark bg rounder corners: https://miro.medium.com/freeze/fit/c/160/160/1*JINdTrpldrRv2LcYpoW0Vg.gif

##### Current TVL:
26M

##### Chain:
Polygon (most TVL) and xDai

##### Short Description (to be shown on DefiLlama):
Programmable Cashflows. Handle subscriptions, salaries, rewards and any composable stream of value, with continuous settlement and per-second netting for extreme capital efficiency.

##### Token address and ticker if any:
Superfluid lets users create Supertokens, to create streams of DAIx or WETHx for DAI and WETH

##### Category (Yield/Dexes/Lending/Minting/Assets/Insurance/Options/Indexes/Staking) *Please choose only one:
MoneyStreams

##### methodology (what is being counted as tvl, how is tvl being calculated):
TVL is the total quantity of tokens locked in Super Tokens from Superfluid, on Polygon and xDai (most important being weth, dai, usdc and wbtc, as well as QiDAO and MOCA)

### CODE REVIEW

**Comment by jo-chemla** [comment]
Discussion ongoing on the superfluid discord server : 
 - Some tokens are missing from TVL calculation, specifically those not present on mainnet and only on sidechain polygon/xdai
 - an updated graphql endpoint will be published soon, returning tokens totalSupply among other things.

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,5 @@ historical-data.js
 /.idea
 yarn.lock
 .DS_Store
+projects/pooltogether/index.js
```
Comment (0xngmi) [inline .gitignore:10]: why ignore pooltogether and package-lock?

File: projects/superfluid.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  return balances
+}
+
+const tokensNativeToSidechain = [
+  '0x2bf2ba13735160624a0feae98f6ac8f70885ea61', // xdai FRACTION
+  '0x63e62989d9eb2d37dfdb1f93a22f063635b07d51'  // xdai MIVA 
+]
+
+let selectedTokens = ['USDCx', 'ETHx', 'DAIx', 'WBTCx', 'USDTx', 'SDTx', 'QIx', 'MOCAx']
+selectedTokens = ['WORKx', 'WORK', 'RIC', 'MIVA', 'FRACTION']
```
[the remark this answers is not in the document]
Comment (0xngmi) [reply projects/superfluid.js:117]: I guess this is only used for debugging so it doesn't really matter

File: projects/superfluid.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+async function getChainBalances(allTokens, chain, block, transform = a => a) {
+  // Init empty balances
+  let balances = {};
+
+  // Abi MultiCall to get supertokens supplies
+  const lockedTokensOutput = await sdk.api.abi.multiCall({
+    abi: 'erc20:totalSupply', // abi['totalSupply'],
+    calls: allTokens.map(token => ({
+      target: token.id, 
+      methodName: 'totalSupply',
```
Comment (0xngmi) [inline projects/superfluid.js:53]: methodName and reference aren't actually used by sdk.api.abi.multiCall

File: projects/superfluid.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jo-chemla) [reply projects/superfluid.js:117]: Yes this was to prevent low TVL tokens from polluting when debugging.

**Comment by jo-chemla** [comment]
Thanks for the feedback! 
[Superfluid Project card](https://github.com/DefiLlama/DefiLlama-Adapters/projects/2#card-70190258) can be closed

**Pull request merged**